test/common/api/os_sys_calls_test.cc

```diff
@@ -1,9 +1,68 @@
+#ifdef WIN32
+#include <fcntl.h> // Needed for `O_CREAT` and `O_RDWR` on Win32.
+
+#endif
+
 #include "source/common/api/os_sys_calls_impl.h"
 
+#include "test/test_common/environment.h"
+
 #include "gtest/gtest.h"
 
 namespace Envoy {
 
+// Test happy path for `open`, `pwrite`, `pread`, `fstat`, `close`, `stat` and `unlink`.
+TEST(OsSyscallsTest, OpenPwritePreadFstatCloseStatUnlink) {
```

Comment:
ugh I really want to break these out but you can't really fully test read without write/open, ditto fstat.   so yeah can leave as is but if fully testing close requires tweaking anything let's at least add line breaks between the different sections.  if close is tested we can merge as-is

Reply:
Close is [kind of] tested, in that this new test tests the new path, and existing tests implicitly cover the old path. But it turns out Windows doesn't like to *unlink* a file that was created without explicit windows-specific permissions flags, so that shot the auto-merge. :(